pkg/telemetry/metrics_db_lock.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package telemetry
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/common"
+
+	"github.com/juju/fslock"
+)
+
+const (
+	LocalTanzuCLIMetricsDBFileLock = ".cli_metrics_db.lock"
+	// DefaultMetricsDBLockTimeout is the default time waiting on the filelock
+	DefaultMetricsDBLockTimeout = 10 * time.Minute
+)
+
+var cliMetricDBLockFile string
+
+// cliMetricDBLock used as a static lock variable that stores fslock
+// This is used for interprocess locking of the tanzu cli metrics DB file
+var cliMetricDBLock *fslock.Lock
+
+// cliMetricDBMutex is used to handle the locking behavior between concurrent calls
+// within the existing process trying to acquire the lock
+var cliMetricDBMutex sync.Mutex
+
+// AcquireTanzuMetricDBLock tries to acquire lock to update tanzu cli metrics DB file with timeout
+func AcquireTanzuMetricDBLock() {
+	var err error
+
+	if cliMetricDBLockFile == "" {
+		cliMetricDBLockFile = filepath.Join(common.DefaultCLITelemetryDir, LocalTanzuCLIMetricsDBFileLock)
+	}
+
+	// using fslock to handle interprocess locking
+	lock, err := getFileLockWithTimeOut(cliMetricDBLockFile, DefaultMetricsDBLockTimeout)
+	if err != nil {
+		panic(fmt.Sprintf("cannot acquire lock for Tanzu CLI metrics DB, reason: %v", err))
+	}
+
+	// Lock the mutex to prevent concurrent calls to acquire and configure the cliMetricDBLock
+	cliMetricDBMutex.Lock()
+	cliMetricDBLock = lock
```

Comment:
How about releasing the lock after a certain period of time, such as 5 minutes, regardless of the `ReleaseTanzuMetricDBLock()` function being called!

Reply:
The client code using this `ReleaseTanzuMetricDBLock()`should have a  timeout mechanism to release the lock( calling the `ReleaseTanzuMetricDBLock()` method after timeout.